### PR TITLE
overlay.d: Add 30gcp-udev-rules dracut module

### PR DESCRIPTION
**Pull request by ravanelli**
- Add dracut module for 30gcp-udev-rules;
 - Udpate overlay documentation about it.

### CODE REVIEW

File: overlay.d/30gcp-udev-rules/usr/lib/dracut/modules.d/30gcp-udev-rules/google_nvme_id [hunk 1]
```diff
@@ -0,0 +1,308 @@
+#!/bin/bash
```
Comment (jlebon) [inline overlay.d/30gcp-udev-rules/usr/lib/dracut/modules.d/30gcp-udev-rules/google_nvme_id]: Hmm, isn't the file already in the overlay? You should be able to `inst_simple` it directly from where it is in `/usr/lib/udev`.

File: overlay.d/30gcp-udev-rules/usr/lib/dracut/modules.d/30gcp-udev-rules/module-setup.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+# called by dracut
+check() {
+    return 0
+}
+
+# called by dracut
+depends() {
+    return 0
+}
```
Comment (jlebon) [inline overlay.d/30gcp-udev-rules/usr/lib/dracut/modules.d/30gcp-udev-rules/module-setup.sh]: Minor: this shouldn't be necessary

File: overlay.d/30gcp-udev-rules/usr/lib/dracut/modules.d/30gcp-udev-rules/module-setup.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+# called by dracut
+install() {
+
```
Comment (jlebon) [inline overlay.d/30gcp-udev-rules/usr/lib/dracut/modules.d/30gcp-udev-rules/module-setup.sh]: Super minor: extraneous space

**Review by cgwalters — DISMISSED** [review]

File: overlay.d/30gcp-udev-rules/usr/lib/dracut/modules.d/30gcp-udev-rules/module-setup.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/bash
+# Install 64-gce-disk-removal.rules, 65-gce-disk-naming.rules and
+# google_nvme_id into the initramfs
+
+# called by dracut
+install() {
+    inst_simple google_nvme_id /usr/lib/udev/google_nvme_id
```
Comment (jlebon) [inline overlay.d/30gcp-udev-rules/usr/lib/dracut/modules.d/30gcp-udev-rules/module-setup.sh]: Huh, I would've expected

```suggestion
    inst_simple /usr/lib/udev/google_nvme_id
```

here.

What you have clearly works (I don't see a dracut error in the compose) but I'm confused how. I would think this gets interpreted as asking dracut to install the executable `google_nvme_id` into `/usr/lib/udev/google_nvme_id` but `google_nvme_id` isn't in the `PATH` so it shouldn't find it. Can you sanity-check that the file indeed ends up in there? Alternatively, maybe cleaner to just use the simpler invocation? (But if you've already verified this, then feel free to ship as is).

**Review by jlebon — DISMISSED** [review]

File: overlay.d/30gcp-udev-rules/usr/lib/dracut/modules.d/30gcp-udev-rules/module-setup.sh (hunk 4 above)
Comment (ravanelli) [reply overlay.d/30gcp-udev-rules/usr/lib/dracut/modules.d/30gcp-udev-rules/module-setup.sh]: Yeah it works in the end:
```
$ ls -ls /usr/lib/udev/google_nvme_id
12 -rw-r--r--. 2 root root 9186 Jan  1  1970 /usr/lib/udev/google_nvme_id
```
But let me do it this way to be simpler

**Review by jlebon — APPROVED** [review]

**Pull request merged**